openwisp_firmware_upgrader/base/models.py

```diff
@@ -105,7 +105,7 @@ def __str__(self):
 
     def clean(self):
         # Make sure that ('category__organization', 'os') is unique too
-        if not self.os:
+        if not (self.os and hasattr(self, "category")):
```

Comment:
since `self.category` is going to be accessed anyway below, and `hasattr` causes the attribute to be accessed anyway, I think it may be better to do:

```python
try:
  category = self.category
except ObjectDoesNotExists:
  return
if not self.os:
  return
```
And then below change `self.category` to `category`.

It's a bit longer but more explicit and saves the accessed call which is reused.

Reply:
Yeah, that is much neater. Thanks.